libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java

```diff
@@ -477,4 +455,95 @@ public void run() {
         };
         return scheduler.scheduleAtFixedRate(publishRunnable, 0, sPublishFrequencyInHours, TimeUnit.HOURS);
     }
+
+    /**
+     * One time upgrade steps from older versions to Mobile SDK 8.2. Only for internal use!
+     *
+     * @param account User account.
+     * @param context Context.
+     * @deprecated Will be removed in Mobile SDK 10.0.
+     */
+    public static synchronized void upgradeTo8Dot2(UserAccount account, Context context) {
+        final String filenameSuffix = (account != null) ? account.getCommunityLevelFilenameSuffix()
+                : UNAUTH_INSTANCE_KEY;
+        final File rootDir = context.getFilesDir();
+        final List<File> oldEventFiles = getAllEventFiles(rootDir, filenameSuffix);
+        final List<InstrumentationEvent> oldEvents = new ArrayList<>();
+        for (final File file : oldEventFiles) {
+            final InstrumentationEvent event = fetchEvent(file);
+            if (event != null) {
+                oldEvents.add(event);
+                file.delete();
+            }
+        }
+        final SalesforceAnalyticsManager sfAnalyticsManager = new SalesforceAnalyticsManager(account);
+        sfAnalyticsManager.getEventStoreManager().storeEvents(oldEvents);
+    }
+
+    /*
+     * TODO: Exists only for upgrade steps to 8.2. Remove this in Mobile SDK 10.0.
+     */
+    private static InstrumentationEvent fetchEvent(File file) {
```

Comment:
Fetching the event from the old file. Will be removed in `10.0` when the upgrade steps are removed.